node/subsystem/src/lib.rs

```diff
@@ -71,7 +73,7 @@ pub enum FromOverseer<M> {
 ///   * Subsystems dying when they are not expected to
 ///   * Subsystems not dying when they are told to die
 ///   * etc.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
```

Comment:
I wonder why we derive `PartialEq`, but not `Eq` on some types.
The reason `PartialEq` exists is floating point (NaN != NaN): https://doc.rust-lang.org/std/cmp/trait.PartialEq.html. 

So I think we should also derive `Eq` wherever we derive `Eq`.

Reply:
https://github.com/paritytech/polkadot/pull/1403/commits/5b464b0b4d90e9728faffcfc6d6fcda36d3d97ea